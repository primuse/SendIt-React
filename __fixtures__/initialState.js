export const props = {
  auth: {
    isLoading: false,
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com',
      isadmin: false
    }
  },
  parcel: {
    parcels: [
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    isLoading: false,
    userParcel: {
      id: 2,
      parcelName: 'rice',
      weigth: '30',
      price: '4,000',
      destination: 'Owerri',
      receiver: 'Tiku Okoye',
      senton: '22-03-2019',
      status: 'created'
    }
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ],
    show: false
  },
  user: {
    firstName: 'Tiku',
    lastName: 'Okoye',
    registered: '22-03-2019',
    email: 'cim@gmail.com',
    isadmin: false
  }
};

export const props2 = {
  auth: {
    isLoading: false,
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com',
      isadmin: true
    }
  },
  parcel: {
    parcels: [
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    isLoading: false,
    userParcel: {
      id: 2,
      parcelName: 'rice',
      weigth: '30',
      price: '4,000',
      destination: 'Owerri',
      receiver: 'Tiku Okoye',
      senton: '22-03-2019',
      status: 'created'
    }
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ],
    show: false
  },
  user: {
    firstName: 'Tiku',
    lastName: 'Okoye',
    registered: '22-03-2019',
    email: 'cim@gmail.com',
    isadmin: true
  }
};

export const props3 = {
  parcel: {
    isLoading: true
  }
};
