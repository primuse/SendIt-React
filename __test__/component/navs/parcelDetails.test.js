import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import ParcelDetails from '../../../src/components/navs/parcelDetails.jsx';
import { props, props2 } from '../../../__fixtures__/initialState';

const mockStore = configureStore([thunk]);


describe('Parcel Details component', () => {
  const store = mockStore(props);
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><ParcelDetails user={props.user }/></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('User Parcel Details component', () => {
  const store = mockStore(props);
  let parcelDetails;
  beforeEach(() => {
    parcelDetails = mount(<Provider store={store}>
      <ParcelDetails user={props.user} />
      </Provider>);
  });
  afterEach(() => {
    parcelDetails.unmount();
  });
  it('renders the Parcel Details', () => {
    const h6 = parcelDetails.find('h6');
    expect(h6.length).toEqual(10);
    expect(parcelDetails.find('#parcel')).toBeDefined();
  });
  it('opens the change destination modal', () => {
    const aTag = parcelDetails.find('a').at(0);
    aTag.simulate('click');
  });
  it('opens the cancel parcel modal', () => {
    const aTag = parcelDetails.find('a').at(1);
    aTag.simulate('click');
  });
});

describe('Admin Parcel Details component', () => {
  const store = mockStore(props2);
  let parcelDetails;
  beforeEach(() => {
    parcelDetails = mount(<Provider store={store}>
      <ParcelDetails user={props2.user} />
    </Provider>);
  });
  afterEach(() => {
    parcelDetails.unmount();
  });
  it('renders the Parcel Details', () => {
    const h6 = parcelDetails.find('h6');
    expect(h6.length).toEqual(10);
    expect(parcelDetails.find('#parcel')).toBeDefined();
  });
  it('opens the change location modal', () => {
    const aTag = parcelDetails.find('a').at(0);
    aTag.simulate('click');
  });
  it('opens the change status modal', () => {
    const aTag = parcelDetails.find('a').at(1);
    aTag.simulate('click');
  });
});
