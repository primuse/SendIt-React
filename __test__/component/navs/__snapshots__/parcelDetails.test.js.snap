// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page component renders correctly 1`] = `
<div
  className="dash-cont d-flex mt-40"
  id="parcel"
>
  <div
    id="map"
  >
    <img
      src="../src/asset/imgs/parcel.jpg"
    />
  </div>
  <div
    className="parcel-details p-10"
  >
    <h4
      className="parcel-title text-center m-0 mb-24"
      id="parcel-Id"
    >
      PO
      1
    </h4>
    <div
      className="pb-10"
      id="details"
    >
      <div>
        <h6>
          Name:
        </h6>
        <h6>
          Destination:
        </h6>
        <h6>
          Pickup Location:
        </h6>
        <h6>
          Current Location:
        </h6>
        <h6>
          Addressed to:
        </h6>
        <h6>
          Phone Number
        </h6>
        <h6>
          Price:
        </h6>
        <h6>
          Sent On:
        </h6>
        <h6>
          Delivered On:
        </h6>
        <h6>
          Status:
        </h6>
      </div>
      <div>
        <p />
        <p>
          Owerri
        </p>
        <p />
        <p />
        <p>
          Tiku Okoye
        </p>
        <p />
        <p>
          N
          4,000
        </p>
        <p>
          22-03-2019
        </p>
        <p>
          Not yet delivered
        </p>
        <p>
          created
        </p>
      </div>
    </div>
    <div
      className="parcel-footer-btns"
      id="details_btns"
    >
      <a
        className="btn sm bg-blue mr-1 white"
        data-modal={true}
        data-target="#destination-modal"
        href="#"
      >
        Change Destination
      </a>
      <a
        className="btn sm is-outlined-blue"
        data-modal={true}
        data-target="#cancel-modal"
        href="#"
      >
        Cancel Parcel
      </a>
    </div>
  </div>
</div>
`;
